### PR TITLE
First implementation of 3dsmax software launcher

### PR DESCRIPTION
**Pull request by riobo**

### CODE REVIEW

File: plugins/basic/bootstrap/(zero-config)/bootstrap.ms [hunk 1]
```diff
@@ -0,0 +1,39 @@
+-- Copyright (c) 2017 Shotgun Software Inc.
```
Comment (manneohrstrom) [inline plugins/basic/bootstrap/(zero-config)/bootstrap.ms]: would be great with a comment somewhere (not sure where the right place would be) to explain the meaning of the  `\(path)\` syntax.

File: startup.py [hunk 2]
```diff
@@ -0,0 +1,331 @@
...
+
+        :param list versions: List of strings representing versions
+                              to search for. If set to None, search
+                              for all versions. A version string is
+                              DCC-specific but could be something
+                              like "2017", "6.3v7" or "1.2.3.52".
+        :returns: List of :class:`SoftwareVersion` instances
+        """
+        # First look for executables using the Autodesk Synergy registry.
+        sw_versions = self._synergy_software_versions(versions)
```
Comment (manneohrstrom) [inline startup.py]: any reason we are doing both synergy and registry? We were debating this and after some discussion actually ended up dropping the synergy support from Maya. Maybe we could just do the registry based check for max?

File: startup.py [hunk 3]
```diff
@@ -0,0 +1,331 @@
...
+        load Toolkit and the tk-3dsmaxplus engine when 3dsMax starts.
+
+        :param str exec_path: Path to 3dsMax executable to launch.
+        :param str args: Command line arguments as strings.
+        :param str file_to_open: (optional) Full path name of a file to open on launch.
+        :returns: :class:`LaunchInformation` instance
+        """
+        required_env = {}
+
+        startup_script_path = os.path.join(self.disk_location, "plugins", "basic", "bootstrap", "(zero-config)", "bootstrap.ms")
```
Comment (manneohrstrom) [inline startup.py]: i guess we'll see later on if it makes more sense to keep a single startup file (e.g. in `tk-3dsmaxplus/startup`) or separate ones for classic and plugins!

File: startup.py [hunk 4]
```diff
@@ -0,0 +1,331 @@
...
+                    self.logger.info(
+                        "Found 3dsMax install in '%s' but only versions %s "
+                        "and above are supported" % (exec_path, self.minimum_supported_version)
+                    )
+
+                # Create a SoftwareVersion using the information from executable
+                # path(s) found in default locations.
+                self.logger.debug("Creating SoftwareVersion for executable '%s'." % exec_path)
+                sw_versions.append(SoftwareVersion(
+                    default_version,
```
Comment (manneohrstrom) [inline startup.py]: heads up that we made some breaking changes to the interface last week as part of building nuke and houdini. There is a now a different set of params:

https://github.com/shotgunsoftware/tk-core/blob/develop/zero_config/python/tank/platform/software_launcher.py#L629

File: startup.py (hunk 2 above)
Comment (riobo) [inline startup.py]: Yes, let's drop the synergy support.

File: plugins/basic/python/tk_3dsmaxplus_basic/plugin_bootstrap.py [hunk 5]
```diff
@@ -14,6 +14,14 @@
 from . import constants
 
 
+def load(root_path):
+    """
+    Entry point for plugin loading. Called by the startup.py.
```
Comment (manneohrstrom) [inline plugins/basic/python/tk_3dsmaxplus_basic/plugin_bootstrap.py]: "called by startup.py"

File: python/startup/bootstrap.py [hunk 6]
```diff
@@ -47,36 +98,18 @@ def bootstrap_sgtk():
...
-    file_to_open =  os.environ.get("TANK_FILE_TO_OPEN")
+    file_to_open = os.environ.get("SGTK_FILE_TO_OPEN")
     if file_to_open:
         MaxPlus.FileManager.Open(file_to_open)
 
     # clean up temp env vars
-    for var in ["TANK_ENGINE", "TANK_CONTEXT", "TANK_FILE_TO_OPEN"]:
+    for var in ["SGTK_ENGINE", "SGTK_CONTEXT", "SGTK_FILE_TO_OPEN", "SGTK_LOAD_MAX_PLUGINS"]:
         if var in os.environ:
             del os.environ[var]
```
Comment (manneohrstrom) [inline python/startup/bootstrap.py]: now that this has got multiple callers, i think it's worth pointing out that this is the entry point if you are running from older versions of the multi launch app

File: python/startup/bootstrap.py [hunk 7]
```diff
@@ -47,36 +98,18 @@ def bootstrap_sgtk():
...
 
     # if a file was specified, load it now
-    file_to_open =  os.environ.get("TANK_FILE_TO_OPEN")
+    file_to_open = os.environ.get("SGTK_FILE_TO_OPEN")
     if file_to_open:
         MaxPlus.FileManager.Open(file_to_open)
 
     # clean up temp env vars
-    for var in ["TANK_ENGINE", "TANK_CONTEXT", "TANK_FILE_TO_OPEN"]:
+    for var in ["SGTK_ENGINE", "SGTK_CONTEXT", "SGTK_FILE_TO_OPEN", "SGTK_LOAD_MAX_PLUGINS"]:
```
Comment (manneohrstrom) [inline python/startup/bootstrap.py]: i don't think we can rename these - users who run older versions of the launch app need to be supported!

File: plugins/basic/python/tk_3dsmaxplus_basic/plugin_bootstrap.py (hunk 5 above)
Comment (riobo) [inline plugins/basic/python/tk_3dsmaxplus_basic/plugin_bootstrap.py]: Fixed

File: python/startup/bootstrap.py (hunk 7 above)
Comment (riobo) [inline python/startup/bootstrap.py]: Fixed

**Pull request merged**